packages/bridge/contracts/l1/L1DaiGateway.sol

```diff
@@ -81,51 +100,99 @@ contract L1DaiGateway is L1ArbitrumExtendedGateway {
     emit Closed();
   }
 
-  function createOutboundTx(
-    address _from,
-    uint256 _tokenAmount,
-    uint256 _maxGas,
-    uint256 _gasPriceBid,
-    uint256 _maxSubmissionCost,
-    bytes memory _outboundCalldata
-  ) internal override returns (uint256) {
+  function outboundTransfer(
+    address l1Token,
+    address to,
+    uint256 amount,
+    uint256 maxGas,
+    uint256 gasPriceBid,
+    bytes calldata data
+  ) external payable returns (bytes memory) {
     // do not allow initiating new xchain messages if bridge is closed
     require(isOpen == 1, "L1DaiGateway/closed");
-
-    return
-      sendTxToL2(
-        inbox,
-        counterpartGateway,
-        _from,
-        msg.value, // we forward the L1 call value to the inbox
-        0, // l2 call value 0 by default
-        L2GasParams({
-          _maxSubmissionCost: _maxSubmissionCost,
-          _maxGas: _maxGas,
-          _gasPriceBid: _gasPriceBid
-        }),
-        _outboundCalldata
+    require(l1Token == l1Dai, "L1DaiGateway/token-not-dai");
+
+    // we use nested scope to avoid stack too deep errors
```

Comment:
Return values always take up a spot in the stack, if you use it you can drop the nested scope.
I find the current scope approach cleaner, but throwing the option out there

```solidity
  function outboundTransfer(
       [ ... ]
  ) external payable returns (bytes memory res) {
    // do not allow initiating new xchain messages if bridge is closed
    require(isOpen == 1, "L1DaiGateway/closed");
    require(l1Token == l1Dai, "L1DaiGateway/token-not-dai");

    address from;
    uint256 maxSubmissionCost;
    // res is used to save on the stack
    ( from, maxSubmissionCost, res ) = parseOutboundData(data);
    require(res.length == 0, "L1DaiGateway/call-hook-data-not-allowed");
    TokenLike(l1Token).transferFrom(from, l1Escrow, amount);

    bytes memory outboundCalldata = getOutboundCalldata(l1Token, from, to, amount, res);
    uint256 seqNum = sendTxToL2(
      l2Counterpart,
      from,
      0,
      maxSubmissionCost,
      maxGas,
      gasPriceBid,
      outboundCalldata
    );
```